samcli/lib/utils/tar.py

```diff
@@ -79,9 +79,22 @@ def _validate_destinations_exists(tar_paths: List[Union[str, Path]]) -> bool:
     """
     for file in tar_paths:
         file_path_obj = Path(file)
-        resolved_path = file_path_obj.resolve()
 
-        if file_path_obj.is_symlink() and not resolved_path.exists():
+        try:
+            resolved_path = file_path_obj.resolve()
+        except OSError:
+            # this exception will occur on Windows and will return
+            # a WinError 123
+            LOG.warning("Failed to determine the file on the system")
+            return False
+
+        if file_path_obj.is_dir():
+            # recursively call this method to validate the children are not symlinks to empty locations
+            children = list(file_path_obj.iterdir())
+            if not _validate_destinations_exists(children):
+                # exits early
+                return False
+        elif file_path_obj.is_symlink() and not resolved_path.exists():
             LOG.warning(f"Symlinked file {file_path_obj} -> {resolved_path} does not exist!")
             return False
```

Comment:
There is currently no logic after this `if not _validate_destinations_exists(children)` check . Is there reason to believe there would be more logic in the future? If not we can reduce some nesting here, and get rid of the`else`
```suggestion
        if file_path_obj.is_dir():
            # recursively call this method to validate the children are not symlinks to empty locations
            children = list(file_path_obj.iterdir())
           return _validate_destinations_exists(children):

        if file_path_obj.is_symlink() and not resolved_path.exists():
            LOG.warning(f"Symlinked file {file_path_obj} -> {resolved_path} does not exist!")
            return False
```

Reply:
The reason for the nested `if` statement is to return early only if we found an invalid symlink. Returning true here will stop the loop and not evaluate the other tar paths.

Reply:
Ah yes the loop, my mistake. That makes sense